src/test/groovy/unit/org/pih/warehouse/api/ProductApiTest.groovy

```diff
@@ -0,0 +1,178 @@
+        productId = product.id
+
+        then: "product should not be null"
+        assertThat(productId, is(notNullValue()))
+
+    }
+
+    void "get demand of product"() {
+        given: "api/products/${productId}/demand called"
+        RequestSpecification httpRequest = given().spec(requestSpecification).
+                pathParams("productId", productId)
+
+        when: "we request product demand resource"
+        Response response = httpRequest.get("/api/products/{productId}/demand")
+        JsonPath jsonPathEvaluator = response.jsonPath()
+        def data = jsonPathEvaluator.get("data")
+
+        then:
+        response.then().assertThat().statusCode(200)
+
+//        then: "data should not be null"
+//        Assert.assertNotNull(data)
+//        and: "product should not be null"
+//        Assert.assertNotNull(data.product, "Response should have product object")
+//        and: "product ID should not be null"
+//        Assert.assertEquals(data.product.id, productId, "Product id should match with requested id:${productId}")
+//        and: "demand should not be null"
+//        Assert.assertNotNull(data.demand, "Demand Object should not be null")
+    }
+
+    void "get demand summary of product"() {
+        given:
+        RequestSpecification httpRequest = given().spec(requestSpecification).
+                pathParams("productId", productId)
+
+        when:
+        Response response = httpRequest.get("/api/products/{productId}/demandSummary")
+        // Get JSON Representation from Response Body
+        JsonPath jsonPathEvaluator = response.jsonPath()
+        def data = jsonPathEvaluator.getList("data")
+
+        then:
+        response.then().assertThat().statusCode(200)
+//        Assert.that(is(notNullValue(data)))
+//        and:
+//        Assert.that(is(data.size(), equalTo(0)))
+    }
+
+    void "get summary of product"() {
+        given:
+        RequestSpecification httpRequest = given().spec(requestSpecification).pathParam("productId", productId)
+
+        when: "api/products/${productId}/productSummary called"
+        Response response = httpRequest.get("/api/products/{productId}/productSummary")
+        // Get JSON Representation from Response Body
+        JsonPath jsonPathEvaluator = response.jsonPath()
+        def data = jsonPathEvaluator.get("data")
+
+        then:
+        response.then().assertThat().statusCode(200)
+//        Assert.assertEquals(data.product.id, productId, "Product id should be ${productId}")
+//        and:
+//        Assert.assertEquals(data.location.id, locationId, "Location id should be ${locationId}")
+//        and:
+//        Integer quantityOnHand = data.quantityOnHand ?: 0
+//        Assert.assertEquals(quantityOnHand, 0, "QuantityOnHand should be 0")
+    }
+}
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+class ProductResponse {
+    String id
```

Comment:
Add new line
```suggestion
class ProductResponse {

    String id
```

Reply:
I think there is no need to add an empty line here.
Lately I've been trying to do a research whether the properties should be even separated by an empty line if they don't contain any annotations above, but can't find any good resource about that.
I meant:
```groovy
class ProductResponse {

  String id

  String productCode
 
  String name
}
```
vs
```groovy
class ProductResponse {
  String id
  String productCode
  String name
}
```

Reply:
I am a fan of adding a new line before each property (the first version), but if u think there is no need, so it can stay

Reply:
Can we spend some more time on the substance? This class is going to be removed once we deal with the inconsistences within our JSON responses. We just needed a temporary solution to write these reference implementation tests.